File: backend/app/models/college.py
from __future__ import annotations

from sqlalchemy import JSON, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base, TimestampMixin, UUIDPrimaryKeyMixin

COLLEGE_STATUSES = ("draft", "active", "suspended", "archived")


class College(UUIDPrimaryKeyMixin, TimestampMixin, Base):
    """The primary tenant entity. Every college-owned row references this."""

    __tablename__ = "colleges"

    name: Mapped[str] = mapped_column(String(255), nullable=False)
    slug: Mapped[str] = mapped_column(String(100), unique=True, nullable=False, index=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    logo_url: Mapped[str | None] = mapped_column(Text, nullable=True)
    website_url: Mapped[str | None] = mapped_column(Text, nullable=True)
    email: Mapped[str | None] = mapped_column(String(255), nullable=True)
    phone: Mapped[str | None] = mapped_column(String(50), nullable=True)
    address: Mapped[str | None] = mapped_column(Text, nullable=True)
    city: Mapped[str | None] = mapped_column(String(100), nullable=True)
    state: Mapped[str | None] = mapped_column(String(100), nullable=True)
    country: Mapped[str | None] = mapped_column(String(100), nullable=True)
    postal_code: Mapped[str | None] = mapped_column(String(20), nullable=True)
    timezone: Mapped[str] = mapped_column(String(100), nullable=False, default="Asia/Kolkata")
    default_language: Mapped[str] = mapped_column(String(20), nullable=False, default="en")
    supported_languages: Mapped[list] = mapped_column(JSON, nullable=False, default=lambda: ["en"])
    feature_flags: Mapped[dict] = mapped_column(JSON, nullable=False, default=dict)
    status: Mapped[str] = mapped_column(String(30), nullable=False, default="draft", index=True)

File: frontend/types/college.ts
export type ConfigurationStatus = {
  identity: string;
  branding: string;
  languages: string;
  admissions: string;
  knowledge: string;
  agent: string;
};

export type College = {
  id: string;
  name: string;
  slug: string;
  description: string | null;
  logo_url: string | null;
  website_url: string | null;
  email: string | null;
  phone: string | null;
  city: string | null;
  state: string | null;
  country: string | null;
  timezone: string;
  default_language: string;
  supported_languages: string[];
  feature_flags: Record<string, boolean | string | number | null>;
  status: string;
  configuration_status: ConfigurationStatus;
};

export type CollegeIdentityUpdate = Partial<{
  name: string;
  description: string | null;
  website_url: string | null;
  email: string | null;
  phone: string | null;
  address: string | null;
  city: string | null;
  state: string | null;
  country: string | null;
  postal_code: string | null;
}>;

export type CollegeConfigurationUpdate = Partial<{
  timezone: string;
  default_language: string;
  supported_languages: string[];
  feature_flags: Record<string, boolean | string | number | null>;
  logo_url: string | null;
}>;
